### PR TITLE
Update EdgeIdentity to major version 4.0,  minimum Xcode version to 14.1 and iOS 11.

### PR DESCRIPTION
**Pull request by cacheung**
To Incorporate that as of April 25th 2023, iOS and iPadOS apps submitted to the App Store must be built with Xcode 14.1.(https://developer.apple.com/news/upcoming-requirements/?id=04252023a)

Update EdgeIdentity to major version 4.0,  minimum Xcode version to 14.1 and iOS 11.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -62,6 +75,18 @@ jobs:
       env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+    - name: Upload - AEPEdgeIdentity
+      uses: actions/upload-release-asset@v1
+      if: ${{ github.event.inputs.action_tag == 'yes' }}
+      with:
+        upload_url: ${{ steps.create_release.outputs.upload_url }}
+        asset_path: ./build/AEPEdgeIdentity.xcframework.zip
+        asset_name: AEPEdgeIdentity.xcframework.zip
```
Comment (emdobrin) [inline .github/workflows/release.yml:84]: also include the version in the artifact name:
asset_name: AEPEdgeIdentity-${{ github.event.inputs.tag }}.xcframework.zip

File: Makefile [hunk 2]
```diff
@@ -10,9 +10,9 @@ IOS_SIMULATOR_ARCHIVE_PATH = $(CURR_DIR)/build/ios_simulator.xcarchive/Products/
 IOS_SIMULATOR_ARCHIVE_DSYM_PATH = $(CURR_DIR)/build/ios_simulator.xcarchive/dSYMs/
 IOS_ARCHIVE_PATH = $(CURR_DIR)/build/ios.xcarchive/Products/Library/Frameworks/
 IOS_ARCHIVE_DSYM_PATH = $(CURR_DIR)/build/ios.xcarchive/dSYMs/
-TVOS_SIMULATOR_ARCHIVE_PATH = ./build/tvos_simulator.xcarchive/Products/Library/Frameworks/
+TVOS_SIMULATOR_ARCHIVE_PATH = $(CURR_DIR)./build/tvos_simulator.xcarchive/Products/Library/Frameworks/
 TVOS_SIMULATOR_ARCHIVE_DSYM_PATH = $(CURR_DIR)/build/tvos_simulator.xcarchive/dSYMs/
-TVOS_ARCHIVE_PATH = ./build/tvos.xcarchive/Products/Library/Frameworks/
+TVOS_ARCHIVE_PATH = $(CURR_DIR)./build/tvos.xcarchive/Products/Library/Frameworks/
```
Comment (emdobrin) [inline Makefile:15]: do these paths need to change? I did not see this in core

File: AEPEdgeIdentity.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -891,6 +891,7 @@
 		};
 		D43BAC2025BF4A0C00B08CC0 /* ShellScript */ = {
 			isa = PBXShellScriptBuildPhase;
+			alwaysOutOfDate = 1;
```
Comment (emdobrin) [inline AEPEdgeIdentity.xcodeproj/project.pbxproj:894]: make sure you uncheck the "based on dependency analysis" box in Swift lint action for all xcode targets that use it

File: AEPEdgeIdentity.podspec [hunk 4]
```diff
@@ -1,6 +1,6 @@
 Pod::Spec.new do |s|
   s.name             = "AEPEdgeIdentity"
-  s.version          = "1.2.0"
```
Comment (emdobrin) [inline AEPEdgeIdentity.podspec:3]: Also update the internal extension version and MARKETING_VERSION in the .xcodeproj

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -61,9 +61,9 @@ jobs:
         make archive
         make zip
 
-     - name: Create GH Release
-       id: create_release
-       uses: release-drafter/release-drafter@v5
+    - name: Create GH Release
+      id: create_release
+      uses: release-drafter/release-drafter@v5
```
Comment (addb) [inline .github/workflows/release.yml:67]: Need to combine L64 to L76 as one job. We basically added name and id for the existing step.

File: Makefile (hunk 2 above)
Comment (addb) [reply Makefile:15]: Just wanted to update to use a common pattern, either use `./build` or `${CURR_DIR}/build`. 

@cacheung  remove . after ${CURR_DIR}
` ${CURR_DIR}/build`

File: Makefile [hunk 6]
```diff
@@ -10,9 +10,9 @@ IOS_SIMULATOR_ARCHIVE_PATH = $(CURR_DIR)/build/ios_simulator.xcarchive/Products/
...
 TVOS_ARCHIVE_DSYM_PATH = $(CURR_DIR)/build/tvos.xcarchive/dSYMs/
 
 TEST_APP_IOS_SCHEME = TestApp
```
Comment (addb) [inline Makefile:18]: There is no `zip` step so make zip from github workflow will fail. Also update the makefile as per edge changes for uniformity.

File: Script/test-podspec.sh [hunk 7]
```diff
@@ -68,13 +68,13 @@ swift package generate-xcodeproj
...
   use_frameworks!
-  pod 'AEPCore', '~> 3.7'
-  pod 'AEPIdentity', '~> 3.7'
-  pod 'AEPServices', '~> 3.7'
-  pod 'AEPRulesEngine', '~> 1.0'
+  pod 'AEPCore', '~> 4'
+  pod 'AEPIdentity', '~> 4'
+  pod 'AEPServices', '~> 4'
+  pod 'AEPRulesEngine', '~> 2'
   pod 'AEPEdgeIdentity', :path => '../AEPEdgeIdentity.podspec'
```
Comment (kevinlind) [inline Script/test-podspec.sh:78]: Can you clean up the TODO on line 81 by adding `AEPEdge` to the tvOS podspec now that it supports tvOS?

Comment (emdobrin) [reply Script/test-podspec.sh:79]: here we should also update to ~> 4, let's update all dependencies in the next pr

**Review by emdobrin — APPROVED** [review]
looks good

**Review by emdobrin — APPROVED** [review]
looks good

**Pull request merged**